### PR TITLE
Add JSX, TSX and TSConfig file templates to Node projects

### PR DESCRIPTION
**Pull request by anubmat**
Add TS, JSX and tsconfig templates for Node projects

### CODE REVIEW

File: Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "noImplicitAny": false,
```
Comment (billti) [inline Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json]: For Node.js we should also set the module to CommonJS.

File: Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "noEmitOnError": true,
+    "removeComments": false,
+    "sourceMap": true,
+    "target": "es5"
+  },
+  "exclude": [
+    "node_modules",
+    "wwwroot"
```
Comment (billti) [inline Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json]: This is an ASP.NET thing. I don't think there's any value in excluding wwwroot for Node.js.

File: Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.vstemplate [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<VSTemplate Version="3.0.0" Type="Item" xmlns="http://schemas.microsoft.com/developer/vstemplate/2005">
+  <TemplateData>
+    <Name Package="{2ffe45c4-5c73-493c-b187-f2e955ff875e}" ID="1024"/>
+    <Description Package="{2ffe45c4-5c73-493c-b187-f2e955ff875e}" ID="1025"/>
+    <!-- pulled from the asp.net v5 package -->
```
Comment (billti) [inline Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.vstemplate]: Will this icon still be present if the ASP.NET tools aren't installed then?

File: Nodejs/Product/Nodejs/Templates/Files/EmptyTs/EmptyTs.vstemplate [hunk 4]
```diff
@@ -8,6 +8,6 @@
     <SortOrder>200</SortOrder>
   </TemplateData>
   <TemplateContent>
-    <ProjectItem SubType="Code" TargetFileName="$fileinputname$.$fileinputextension$" ReplaceParameters="true">EmptyTs.ts</ProjectItem>
+    <ProjectItem SubType="Code" ItemType="TypeScriptCompile" TargetFileName="$fileinputname$.$fileinputextension$" ReplaceParameters="true">EmptyTs.ts</ProjectItem>
```
Comment (billti) [inline Nodejs/Product/Nodejs/Templates/Files/EmptyTs/EmptyTs.vstemplate]: Was this a bug fix you've included? What impact does this change have?

Comment (anubmat) [inline Nodejs/Product/Nodejs/Templates/Files/EmptyTs/EmptyTs.vstemplate]: No this is not a bug fix, but it was always specified in our TS templates, so far, I didn't see any difference. So I'll remove this change from both ts and tsx template files.

File: Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.vstemplate (hunk 3 above)
Comment (anubmat) [inline Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.vstemplate]: Yes, I checked with just Cordova and UWP installed.. and it works.. In TS repo, we always get tsconfig icon from this package.

File: Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json (hunk 2 above)
Comment (anubmat) [inline Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json]: Ok will remove..,

File: Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json (hunk 1 above)
Comment (anubmat) [inline Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json]: ok, do you think it's  good idea to include "compile on save" by default as well?

Comment (anubmat) [inline Nodejs/Product/Nodejs/Templates/Files/TypeScriptTsConfig/tsconfig.json]: Never mind, "compile on save" is not needed to be added... It is on by default..

**Pull request merged**